common/src/gas_provider/mod.rs

```diff
@@ -235,6 +258,9 @@ pub trait Provider {
     /// Type representing a quantity of value.
     type Balance;
 
+    /// Type representing a quantity of token balance.
+    type Funds;
```

Comment:
For the sake of readability I would suggest to rename this field to `ValueBalance` and the former one to `GasBalance`.